qiime2/core/cache.py

```diff
@@ -520,6 +533,18 @@ def __exit__(self, *args):
         _CACHE.cache = self.previously_entered_cache
         self.cache.named_pool = None
 
+    def _get_process_pool_name(self):
+        """Gets the necessary info to create/identify a process pool for this
+        process
+        """
+        pid = os.getpid()
+        user = getpass.getuser()
```

Comment:
@Oddant1 I think this will run into the same issue. We should create some helper to create the "user" which everything in this class can use.

Reply:
Yep good catch. Didn't even think that maybe we called getpass.getuser more than once.